examples/NFD/udp_flood_mitigation/UDPFloodMitagation.cpp

```diff
@@ -0,0 +1,359 @@
+        this->field_value["dip"] = new IP(des_addr, 32);
+        this->field_value["tag"] = new int(0);
+        this->field_value["iplen"] = new int(totallength);
+        short protocol = ntohs(*(short *)(ip_header + 9));
+        this->field_value["UDP"] = new int((protocol == IPPROTO_UDP) ? 1 : 0);
+
+        /*TCP layer*/
+        TCPHdr *tcph = (TCPHdr *)(ip_header + ip_header_length);
+        this->field_value["sport"] = new int(ntohs(tcph->th_sport));
+        this->field_value["dport"] = new int(ntohs(tcph->th_dport));
+        /*URG ACK PSH RST SYN FIN*/
+        this->field_value["flag_fin"] = new int(ntohs(tcph->th_flags) & TH_FIN);
+        this->field_value["flag_syn"] = new int(ntohs(tcph->th_flags) & TH_SYN);
+        this->field_value["flag_ack"] = new int(ntohs(tcph->th_flags) & TH_ACK);
+}
+void
+Flow::clean() {
+        /*Encoding*/
+        struct ether_header *eth_header;
+        eth_header = (struct ether_header *)this->pkt;
+        int ethernet_header_length = 14; /* Doesn't change */
+        int ip_header_length;
+        int tcp_header_length;
+        u_char *ip_header;
+        ip_header = (u_char *)eth_header + ethernet_header_length;
+        ip_header_length = ((*ip_header) & 0x0F);
+        ip_header_length = ip_header_length * 4;
+
+        *((int *)(ip_header + 12)) = htonl(*((int *)this->field_value["sip"]));
+        *((int *)(ip_header + 16)) = htonl(*((int *)this->field_value["dip"]));
+        // this->field_value["iplen"] = new int(totallength);
+
+        /*TCP layer*/
+        TCPHdr *tcph = (TCPHdr *)(ip_header + ip_header_length);
+        tcph->th_sport = u_short(*((int *)this->field_value["sport"]));
+        tcph->th_dport = u_short(*((int *)this->field_value["dport"]));
+}
+
+int
+process(Flow &f);
+Flow f_glb;
+long int _counter = 0;
+long int _drop = 0;
+unordered_map<string, int> F_Type::MAP = unordered_map<string, int>();
+unordered_map<string, int> F_Type::MAP2 = unordered_map<string, int>();
+
+struct timeval begin_time;
+struct timeval end_time;
+
+void
+_init_() {
+        (new F_Type())->init();
+}
+int _t1 = 100;
+int _t2 = 1;
+int _t3 = 1;
+int _t4 = 1;
+int _t5 = 1;
+int _t6 = 1;
+int _t7 = 1;
+int _t8 = 1;
+State<unordered_map<IP, int>> udpcounter(*(new unordered_map<IP, int>()));
+State<unordered_map<IP, int>> udpflood(*(new unordered_map<IP, int>()));
+State<int> threshold(_t1);
+
+int
+process(Flow &f) {
+        if (((*(int *)f["UDP"]) == _t2) &&
+            (udpflood[f][(*(IP *)f["sip"])] != _t3 && udpcounter[f][(*(IP *)f["sip"])] != threshold[f])) {
+                udpcounter[f][(*(IP *)f["sip"])] = udpcounter[f][(*(IP *)f["sip"])] + _t4;
+        } else if (((*(int *)f["UDP"]) == _t5) &&
+                   (udpflood[f][(*(IP *)f["sip"])] != _t6 && udpcounter[f][(*(IP *)f["sip"])] == threshold[f])) {
+                return -1;
+                udpflood[f][(*(IP *)f["sip"])] = _t7;
+        } else if ((*(int *)f["UDP"]) != _t8) {
```

Comment:
What is this branch for

Reply:
the empty branch means that we do no operation on the packets but just pass it.